firebase-firestore/src/main/java/com/google/firebase/firestore/util/ListenerRegistrationImpl.java

```diff
@@ -33,6 +58,119 @@
   /** The event listener for the query that raises events asynchronously. */
   private final ExecutorEventListener<ViewSnapshot> asyncEventListener;
 
+  static class CallbackList {
+    void run() {
+      for (Runnable callback : callbacks) {
+        if (callback != null) {
+          callback.run();
+        }
+      }
+    }
+
+    void add(Runnable callback) {
+      callbacks.add(callback);
+    }
+
+    private final List<Runnable> callbacks = new ArrayList<>();
```

Comment:
optional nit: in Java data members usually go at the top of the class not in the very end(here and below), feel free to ignore if this is a common convention in firestore's codebase

Reply:
done.